### PR TITLE
feat(appsync): add ApiCache L2 construct and addCache method

### PR DESCRIPTION
**Pull request by TonySherman**
### Issue # (if applicable)
#31160
Closes #<issue number here>.

### Description of changes
Adds L2 Construct for ApiCache on AppSync graphQL Apis
and a method to add a cache to existing graphQL Api constructs.

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/aws-appsync/lib/api-cache.ts [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+export interface ApiCacheBaseProps {
+  /**
+   * Caching behavior
+   */
+  readonly apiCachingBehavior: CacheBehavior;
+
+  /**
+   * At-rest encryption flag for cache
+   *
+   * @default - idk
```
Comment (TonySherman) [inline packages/aws-cdk-lib/aws-appsync/lib/api-cache.ts:82]: I could not find the default behavior of the various optional properties for this resource

Comment (TonySherman) [reply packages/aws-cdk-lib/aws-appsync/lib/api-cache.ts]: verified these by deploying without setting the values.

File: packages/aws-cdk-lib/aws-appsync/lib/api-cache.ts [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+  /**
+   * The GraphQL API ID.
+   */
+  readonly apiId: string;
+}
+
+/**
+ * Cache for a GraphQL API
+ */
+export class ApiCache extends Construct {
```
Comment (pahud) [inline packages/aws-cdk-lib/aws-appsync/lib/api-cache.ts:121]: L2 construct should extends `Resource` implements `IApiCache`. You should export `IApiCache` as well.

File: packages/aws-cdk-lib/aws-appsync/lib/api-cache.ts [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+}
+
+/**
+ * Properties for an Api Cache
+ */
+export interface ApiCacheProps extends ApiCacheBaseProps {
+  /**
+   * The GraphQL API ID.
+   */
+  readonly apiId: string;
```
Comment (pahud) [inline packages/aws-cdk-lib/aws-appsync/lib/api-cache.ts:115]: I think it should be

```
readonly api: IGraphqlApi;
```

**Review by pahud — DISMISSED** [review]
If you are implementing the ApiCache L2, I suggest you first read the design guidelines [here](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md) and read other existing L2 constructs to learn the patterns.

Generally, if you are creating a L2 construct called `Foo`. You should:

1. Define `IFoo` interface for it that extends `IResource`.
2. Define `Foo` construct class that extends `Resources` implements `IFoo`
3. Define `FooProps` as the construct props for `Foo`

In addition to the design guides and existing L2 constructs, I would recommend you read [this blog post](https://community.aws/content/2kcTpCDCTOTYOY8UoWGGpttA6t8/building-agentic-code-interpreter-with-aws-cdk) in which I shared how I built an Agent L2 construct.

**Comment by pahud** [comment]
Some suggested changes here:

- Validation for Ttl property:

1. The CloudFormation documentation specifies that the Ttl property should be a number between 1 and 3,600 (inclusive). You can add some validation to ensure that the provided ttl value is within this range.
You can add a check in the constructor of the ApiCache class to validate the ttl value:

```ts
constructor(scope: Construct, id: string, props: ApiCacheProps) {
  super(scope, id);

  if (props.ttl < 1 || props.ttl > 3600) {
    throw new Error('Ttl value must be between 1 and 3,600 seconds');
  }

  new CfnApiCache(this, 'Resource', {
    // Rest of the code
  });
}
```

**Comment by TonySherman** [comment]
> Some suggested changes here:
> 
> * Validation for Ttl property:
> 
> 1. The CloudFormation documentation specifies that the Ttl property should be a number between 1 and 3,600 (inclusive). You can add some validation to ensure that the provided ttl value is within this range.
>    You can add a check in the constructor of the ApiCache class to validate the ttl value:
> 
> ```ts
> constructor(scope: Construct, id: string, props: ApiCacheProps) {
>   super(scope, id);
> 
>   if (props.ttl < 1 || props.ttl > 3600) {
>     throw new Error('Ttl value must be between 1 and 3,600 seconds');
>   }
> 
>   new CfnApiCache(this, 'Resource', {
>     // Rest of the code
>   });
> }
> ```

Thanks! Sorry for the delay on updates to this work. Got sidetracked with some other stuff and I'll be getting back to this shortly.

**Pull request closed without merging**